[Writing Tools] No Writing Tools popover in WebKitLegacy apps

rr-codes opened the pull request:
#### dc0791d45d518ff51ca0b3f757011ab90f053e84
<pre>
[Writing Tools] No Writing Tools popover in WebKitLegacy apps
<a href="https://bugs.webkit.org/show_bug.cgi?id=276840">https://bugs.webkit.org/show_bug.cgi?id=276840</a>
<a href="https://rdar.apple.com/126112513">rdar://126112513</a>

Reviewed by Wenson Hsieh.

Writing Tools is not supported in WebKitLegacy, so remove the context menu item.

* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(createMenuItem):
(-[WebHTMLView writingToolsBehavior]):
(-[WebHTMLView providesWritingToolsContextMenu]):

Canonical link: <a href="https://commits.webkit.org/281158@main">https://commits.webkit.org/281158@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/07b1a6b3d591609960daa0b70eabe618d39b4e4b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58952 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/38281 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/11444 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/62585 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/9396 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45932 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/9598 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/47669 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6689 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60983 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35813 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50986 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/28527 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/32538 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8278 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/8400 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/54490 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/8558 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/64285 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2865 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/8518 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54993 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2875 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/51014 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/55096 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2402 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8811 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34110 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/35194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/36279 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/34940 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
```diff
@@ -7141,6 +7147,22 @@ - (BOOL)_findString:(NSString *)string options:(WebFindOptions)options
+    return PlatformWritingToolsBehaviorNone;
+}
+
+- (BOOL)providesWritingToolsContextMenu
```
This could use some clarification (with a comment), because we're saying "no Writing Tools" but the code appears to be saying "I provide a Writing Tools menu item" 😅

rr-codes committed:
[Writing Tools] No Writing Tools popover in WebKitLegacy apps
https://bugs.webkit.org/show_bug.cgi?id=276840
rdar://126112513

Reviewed by Wenson Hsieh.

Writing Tools is not supported in WebKitLegacy, so remove the context menu item.

* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(createMenuItem):
(-[WebHTMLView writingToolsBehavior]):
(-[WebHTMLView providesWritingToolsContextMenu]):

Canonical link: https://commits.webkit.org/281158@main
